### PR TITLE
Join implmentation

### PR DESCRIPTION
**Pull request by IMarnild**
JoiN vIEw + JoIn viEwmoDel (in WelCoMeVM) + LoBby vIeW + LOBbY ViEwmOdel:
JOIn wOrKs NoW

### CODE REVIEW

File: PlanningPoker.App/Models/SessionClient.cs [hunk 1]
```diff
@@ -121,5 +129,20 @@ public async Task<bool> KickUser(string sessionKey, int userId)
 
             return response.IsSuccessStatusCode;
         }
+
+        public UserStateResponseDTO JoinMock(string sessionkey, UserCreateDTO user)
```
Comment (oliverjuhl) [inline PlanningPoker.App/Models/SessionClient.cs]: Delete

File: PlanningPoker.App/PlanningPoker.App.csproj [hunk 2]
```diff
@@ -12,6 +12,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNet.WebApi.Client" Version="5.2.7" />
     <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
+    <PackageReference Include="OpenJobScheduler" Version="1.0.1" />
```
Comment (oliverjuhl) [inline PlanningPoker.App/PlanningPoker.App.csproj]: Er det til dit threading?

File: PlanningPoker.App/ViewModels/WelcomeViewModel.cs [hunk 3]
```diff
@@ -70,5 +68,76 @@ public async Task<bool> ExecuteLoginCommand()
...
+            catch (Exception e)
+            {
+                this.Connection = false;
+                this.Key = string.Empty;
+                Debug.WriteLine("No session with that key exists.");
+                Debug.WriteLine(e.StackTrace);
+            }
+        }
+
+        private void JoinSessionMock()
```
Comment (oliverjuhl) [inline PlanningPoker.App/ViewModels/WelcomeViewModel.cs]: Delete

File: PlanningPoker.App/Views/SessionCreation/CreateSession.xaml.cs [hunk 4]
```diff
@@ -30,7 +30,7 @@ private void CreateSessionClicked(object sender, EventArgs e)
         {
             this.itemsViewModel.CreateSessionCommand.Execute(null);
             this.Navigation.PushModalAsync(
-                new Lobby());
+                new Lobby("42"));
```
Comment (oliverjuhl) [inline PlanningPoker.App/Views/SessionCreation/CreateSession.xaml.cs]: Delete

File: PlanningPoker.App/ViewModels/WelcomeViewModel.cs [hunk 5]
```diff
@@ -4,46 +4,44 @@ namespace PlanningPoker.App.ViewModels
...
-            this.publicClientApplication = publicClientApplication;
-            this.settings = settings;
-            this.BaseTitle = "Login";
-            this.LoginCommand = new Command(async () => await this.ExecuteLoginCommand());
-            this.JoinCommand = new Command(async () => await this.ExecuteJoinCommand());
-        }
+        public bool Loading { get; set; }
 
-        public ICommand LoginCommand { get; }
+        public bool Connection { get; set; }
```
Comment (oliverjuhl) [inline PlanningPoker.App/ViewModels/WelcomeViewModel.cs]: delete

File: PlanningPoker.App/ViewModels/WelcomeViewModel.cs [hunk 6]
```diff
@@ -70,5 +68,76 @@ public async Task<bool> ExecuteLoginCommand()
...
+
+        private async Task ExecuteJoinCommand()
+        {
+            if (this.Loading)
+            {
+                return;
+            }
+
+            this.Loading = true;
+            this.Connection = false;
```
Comment (oliverjuhl) [inline PlanningPoker.App/ViewModels/WelcomeViewModel.cs]: delete

File: PlanningPoker.App/PlanningPoker.App.csproj (hunk 2 above)
Comment (IMarnild) [reply PlanningPoker.App/PlanningPoker.App.csproj]: YES!
 :)

**Pull request merged**